pkg/aggregateeventrecorder/aggregateeventrecorder.go

```diff
@@ -32,6 +32,7 @@ type tempData struct {
 	allowedProtocols         []string
 	attemptedProtocolSet     map[string]struct{}
 	retrievalAttempts        map[string]*RetrievalAttempt
+	metadata                 map[string]interface{}
```

Comment:
can we simplify this to `map[string]string` ?

Reply:
Yeah, after discussing with Rod yesterday during our standup, we agreed to change this to a map of string key to string values to ensure that values can always be marshalled.